pip/req/req_install.py

```diff
@@ -231,27 +237,23 @@ def build_location(self, build_dir):
         # FIXME: Is there a better place to create the build_dir? (hg and bzr
         # need this)
         if not os.path.exists(build_dir):
+            logger.debug('Creating directory %s', build_dir)
             _make_build_dir(build_dir)
         return os.path.join(build_dir, name)
 
     def correct_build_location(self):
         """If the build location was a temporary directory, this will move it
-        to a new more permanent location"""
+        to a new more permanent location
+        Move self._temp_build_dir to self._ideal_build_dir/self.name
+        """
         if self.source_dir is not None:
             return
         assert self.req is not None
         assert self._temp_build_dir
+        assert self._ideal_build_dir
         old_location = self._temp_build_dir
-        new_build_dir = self._ideal_build_dir
-        del self._ideal_build_dir
-        if self.editable:
-            name = self.name.lower()
-        else:
-            name = self.name
-        new_location = os.path.join(new_build_dir, name)
-        if not os.path.exists(new_build_dir):
-            logger.debug('Creating directory %s', new_build_dir)
-            _make_build_dir(new_build_dir)
```

Comment:
This seems to indicate my PR needs more comments :-/
Cf https://github.com/pypa/pip/pull/2506#issuecomment-78033123 which I might need to rewrite and add as comment, maybe as `correct_build_location` docstring ?